tensorflow/c/tf_tensor.h

```diff
@@ -45,6 +45,24 @@ limitations under the License.
 extern "C" {
 #endif
 
+// Macro used to calculate struct size for maintaining ABI stability across 
+// different struct implementations. 
+#ifndef TF_OFFSET_OF_END
+#define TF_OFFSET_OF_END(TYPE, MEMBER) (offsetof(TYPE, MEMBER) + \
+    sizeof(((TYPE *)0)->MEMBER))
+#endif // TF_OFFSET_OF_END
+
+// Allocator Attributes used for tensor allocation. 
+typedef struct TF_AllocatorAttributes { 
```

Comment:
What's the value of exposing this struct here instead of accessor methods? Do we expect users will need to create and modify many instances of it (so the allocations will become expensive)?

Reply:
This struct will be filled by user and passed to TF_AllocateTemp (to be added).
As I understand, accessors would only work if user gets data from TF but its the opposite here.

Reply:
So should this have one of the ABI extensions or will we just v2 it when we need to change it?

Reply:
We can add `void* ext`.
It does have a `struct_size` attribute and we can start checking its value once we add more fields. That way we don't need to add a v2 version of the struct.

Reply:
Hi Alex, looks like this has `struct_size`. So I think it can be extended in an ABI safe manner by appending members to the end of the struct.